src/native/util/os_darwin.cpp

```diff
@@ -43,15 +43,18 @@ get_current_uid()
 const uid_t ThreadScope::orig_uid = getuid();
 const gid_t ThreadScope::orig_gid = getgid();
 const std::vector<gid_t> ThreadScope::orig_groups = get_process_groups();
+int ThreadScope::passwd_buf_size = -1;
 
 static int
 get_supplemental_groups_by_uid(uid_t uid, std::vector<gid_t>& groups)
 {
-    // getpwuid will only indicate if an error happened by setting errno. set it to 0, so will know if there is a change
-    errno = 0;
-    struct passwd* pw = getpwuid(uid);
-    if (pw == NULL) {
-        if (errno == 0) {
+    std::unique_ptr<char[]> buf(new char[ThreadScope::passwd_buf_size]);
+    struct passwd pwd;
+    struct passwd *getpwnam_res;
```

Comment:
initialize pointer to null.
also there is no real need to change the name to be long and anoying, it can stay pw.